commit cc78d770439cfc1ae8ec0c802e5fcb5eae25979f
Author: Nicolai Hähnle <nicolai.haehnle@amd.com>
Date:   Sun Oct 22 17:39:04 2017 +0200

    mesa: flush and wait after creating a fallback texture
    
    Fixes non-deterministic failures in
    dEQP-EGL.functional.sharing.gles2.multithread.simple_egl_sync.images.texture_source.teximage2d_render
    and others in dEQP-EGL.functional.sharing.gles2.multithread.*
    
    Reviewed-by: Marek Olšák <marek.olsak@amd.com>

diff --git a/src/mesa/main/texobj.c b/src/mesa/main/texobj.c
index 71c6f81323..02c4767788 100644
--- a/src/mesa/main/texobj.c
+++ b/src/mesa/main/texobj.c
@@ -1051,6 +1051,11 @@ _mesa_get_fallback_texture(struct gl_context *ctx, gl_texture_index tex)
       assert(texObj->_MipmapComplete);
 
       ctx->Shared->FallbackTex[tex] = texObj;
+
+      /* Complete the driver's operation in case another context will also
+       * use the same fallback texture. */
+      if (ctx->Driver.Finish)
+         ctx->Driver.Finish(ctx);
    }
    return ctx->Shared->FallbackTex[tex];
 }
